.github/workflows/release.yml

```diff
@@ -49,18 +45,40 @@ env:
 
 jobs:
   prepare-release:
+    environment: release
     name: "Prepare release"
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
```

Comment:
`id-token` is necessary for the temporary token, `contents` in order to push the commit and tag.